core/math/src/main/java/org/arakhne/afc/math/geometry/d2/ai/Rectangle2ai.java

```diff
@@ -375,7 +375,7 @@ default double getDistanceSquared(Point2D<?, ?> pt) {
 		} else {
 			dy = 0;
 		}
-		return dx * dx + dy * dy;
+		return (double) (dx * dx + dy * dy);
```

Comment:
This change will be reverted since it causes the ECJ's "unecessary cast" warning. This warning indicates that a dynamic cast operation could be removed for making the code more efficient.